Make Swarm constructor have BlockChain<T> (#324)
planetarium/libplanet: CHANGES.md

        Instead, a newly added static class `StoreExtension` provides
        an extension method of the same name.
     -  Added `IStore.IterateStateReferences()` method.
 -  `Swarm` constructor became to receive `BlockChain<T>`.  [[#324]]

>  -  `Swarm` became to have type parameter `T` that reprsents action type as

Reviewer: A typo: “reprsents.”  Also it would be better:

```suggestion
 -  `Swarm` became to have type parameter `T` to represent an action type as
```

    like as `BlockChain<T>`.  [[#324]]

>  -  Methods that take `BlockChain<T>` as a parameter in `Swarm` no longer take

Reviewer: ```suggestion
 -  Methods  in `Swarm` that had taken a parameter of `BlockChain<T>` type
```

>     `BlockChain<T>`. And they are no longer the generic methods.  [[#324]]

Reviewer: ```suggestion
    became to neither longer take `BlockChain<T>` nor a generic method.
    Because the `Swarm` constructor takes it instead.  [[#324]]
```

        Instead, a newly added static class `StoreExtension` provides
        an extension method of the same name.
     -  Added `IStore.IterateStateReferences()` method.
 -  `Swarm` constructor became to receive `BlockChain<T>` and have a type

>     parameter `T`.  [[#324]]

Reviewer 2: Actually, `Swarm`(not constructor, but class) became to have a type parameter. how about write them separately?

Author: I separated the changelog.

>  -  `Swarm` became to have type parameter `T`.  [[#324]]

Reviewer 2: it would be better if there is mention about `T`.

```suggestion
 -  `Swarm` became to have type parameter `T` that reprsents action type as like as `BlockChain<T>`.  [[#324]]
```
